pkg/signals/signals_test.go

```diff
@@ -0,0 +1,72 @@
+//go:build !windows
+// +build !windows
+
+// Copyright 2022 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package signals
+
+import (
+	"os"
+	"syscall"
+	"testing"
+	"time"
+)
+
+var (
+	settleTime       = 100 * time.Millisecond
+	fatalWaitingTime = 15 * time.Second
+)
+
+func waitSig(t *testing.T, c <-chan os.Signal, sig os.Signal) {
+	t.Helper()
+	start := time.Now()
+	timer := time.NewTimer(settleTime / 10)
+	defer timer.Stop()
+
+	for time.Since(start) < fatalWaitingTime {
+		select {
+		case s := <-c:
+			if s == sig {
+				return
+			}
+		case <-timer.C:
+			timer.Reset(settleTime / 10)
+		}
+	}
+	t.Fatalf("Timeout after %v waiting for %v", fatalWaitingTime, sig)
+}
+
+func TestGenerateStopSignal(t *testing.T) {
+	GenerateStopSignal()
+	waitSig(t, notifyCh, syscall.SIGTERM)
+	select {
+	case s := <-notifyCh:
+		t.Errorf("Unexpected signal %v", s)
+	default:
+		// Nothing to read - good
+	}
+	// Reset channel
+	notifyCh = make(chan os.Signal, 2)
```

Comment:
This won't be executed if `t.Fatal` in `waitSig` is called.

Reply:
Move it to `TestRegisterSignalHandlers`.

Reply:
An unit test should not leave side effect after it finishes. So instead of re-initializing the global varible in another test, the test itself should recreate the channel. It could do it with defer to guarantee it will be executed.

Reply:
Done.